### DISCUSSION TITLE
Audio Attachments

### DISCUSSION THREAD

**Discussion by ConstantineFrol** [apache/couchdb#4231] [Q&A]
Hi, I am new to CouchDB, get familiar with it not so long ago, during my Honors degree in Software Development, and I am meeting one problem with NoSQL for example in Couch DB you can attach files, all types of files. since I am working on my project which includes CouchDB I am trying to attach a .mp3 file, and this process went smoothly, but when I am trying to get this file back from CouchDB. I am getting lots of errors, and to solve them I am browsing Google again and again, and I see there are examples of how to work with .jpeg format, but not much info about other formats. Or is it because CouchDB is not that good for audio files or it's just me too fussy?
![Untitled](https://user-images.githubusercontent.com/14143437/196267385-809b6f30-e19b-4f34-b2d0-ad3e5b0ccbee.png)

**Comment by nickva** [comment]
@ConstantineFrol it should work. Just make sure when you upload them using the correct content type, or as opposed to `text/...` or `application/json`. In that case CouchDB might try to re-compress them with gzip which won't make sense for compressed audio files.

The list of compressible content types can be adjusted with a configuration setting: https://docs.couchdb.org/en/3.2.2-docs/config/misc.html#attachments/compressible_types

Also make sure you try the latest release (3.2.2) as older releases had a few issues attachments which we had fixed since.

**Reply by ConstantineFrol** [reply]
Good evening [nickva](https://github.com/nickva) Thanks for your reply to my problem, I just edited my question by adding a screenshot of my document in CouchDB. for this I am using Python, I know from Documentation that needs to use get_attachement(_id,_rev, file_name)
but as a response, I am getting class 'CouchDB.client.ViewResults'

**Reply by nickva** [reply]
That looks good. I can't tell why the ViewResults response is coming back. It could be that the response is a generator and you'd have to iterate over it to retrieve the responses.

There could be a case of a name encoding as the file name has spaces and there is a chance it could be related to encoding the spaces properly.

Try to perform a simple curl request on the command line and see what happens. Just to see if it's a client issues or a backend issue.

**Reply by ConstantineFrol** [reply]
Hi [nickva](https://github.com/nickva), sorry for taking your time, but is there any other way to download this file to any local directory?
using curl it returns me a JSON format file.

**Reply by ConstantineFrol** [reply]
After a while, I was able to store the result as a .mp3 file using the curl command:
curl -X GET http://localhost:5984/music_box/doc_id/file_name.mp3>~/Documents/test.mp3
and it worked.
a huge thanks for your time and support
![it worked](https://user-images.githubusercontent.com/14143437/196286485-84342bf7-f6a9-460b-a505-e7716fec9e96.png)

**Reply by nickva** [reply]
Oh no problem at all! Thanks for reaching out!
